storage/docs/rest.md

```diff
@@ -1,69 +1,117 @@
 # REST Api
 
-## Store a file
+## URL scheme
+
+The REST Api uses the following scheme for urls
+
+``` /files/<namespace>/<key> ```
+
+The '<namespace>' component cannot include the the '/' character, while the '<key>' can include zero
+or more directories e.g:
+* ``` /files/libero/foo.txt ```
+* ``` /files/libero/baz/bar/foo.txt ```
+
+The key is unique within the namespace and provides a mapping between the storage entity and the provider entity (e.g. S3).
+
+## Storing a file
+
+To store a file with a namespace of ```libero``` and a key of ```directory/file.pdf```, the following request should be
+used
 
 ```
-PUT /files/namespace/directory/file.ext
+PUT /files/libero/directory/file.pdf
+
 Content-Type: application/pdf
 If-Match: <etag_value>|*
-Libero-file-tags: "key=value,key2=value2,filename=bar.pdf"
+Libero-file-tags: "filename=original-file.pdf,key1=value=1"
+```
 
+The Server should respond with:
+
+```
 Response:
 Code: 201
-Link: "<http://url/foo/bar.pdf>;rel=signed"
+Link: "<http://storage-url/path/to/file>;rel=public"
 Last-Modified: 2019-08-19T14:40:04.123456
+Libero-file-id: bd1c9a15-bc18-4151-a4eb-9f45cedb9700
 ETag: <etag>
 ```
 
-## Get a file
+When uploading for the first time, the ```If-Match``` header should be set to '*'. If the ```If-Match``` fails,
+then a 412 (Precondition Failed) should be returned.
```

Comment:
Could make it required and return a [`428 Precondition Required`](https://tools.ietf.org/html/rfc6585#section-3) if missing?

Reply:
isn't 428 for when If-Match header is missing? maybe i understood it wrong...

Reply:
Yeah, that's what I meant. (Safer writes, but depends if they should be modifiable; refs https://github.com/libero/shared-services-experiment/pull/37#discussion_r317096939).

Reply:
Do we need this header for the demo?